Lib/argparse.py

```diff
@@ -2565,8 +2561,7 @@ def _get_values(self, action, arg_strings):
     def _get_value(self, action, arg_string):
         type_func = self._registry_get('type', action.type, action.type)
         if not callable(type_func):
-            msg = _('%r is not callable')
-            raise ArgumentError(action, msg % type_func)
+            raise RuntimeError('%r is not callable' % (type_func,))
```

Comment:
Why not using a `TypeError` or `ValueError` instead?

Reply:
TypeError would be appropriate exception. But this check is already performed in `add_argumen()`, and it raises ValueError, so ValueError could be more consistent. On other hand, this check will not fail in normal circumstances (the corresponding test has comment "Someone got crazy and did this"), so RuntimeError is an appropriate type for unexpected error.